### PR TITLE
P4-3944: use Redis for short TTL pm download links

### PR DESCRIPTION
**Pull request by baracudda**
# For the Reviewer
- [ ] Code review complete
- [ ] Testing Complete
- [ ] Quality ORT App Documentation Updated (your name is in the Validator square for this feature)

When this is complete, you should approve the PR via github.

# For the Reviewee

## Summary
Replace fixed container UUIDs with Redis UUIDs with short TTL for use with public download PM links. Created an ENV var with special links for PostOffice use that requires auth creds to use.

#### Release Note
Fixed: Downloading PM will not result in a timeout half the time.

#### Breaking Changes
None.

## Quality Assurance

You have gathered the following items:
- [ ] This PR is tagged with a Release Milestone
- [ ] You have a log message clearly identifying when this feature is **working successfully**
- [ ] You have a log message clearly identifying when this feature is **failing**
- [ ] You added a PR against [p4-alerting](https://github.com/istresearch/p4-alerting/) to trigger based on the failure condition above

Given all of the items above, you have updated your Application ORT at the following locations:
- **Features and Alerting**: <link to app ORT>Required.
- **P4 Alerting**: <link to p4-alerting PR>Required.

## Testing and Verification

*Steps to test your application for someone not familiar with it.* Required.

1. Do this: `$ run-this.sh`
2. Look for this.
3. Clean up with this command

### CODE REVIEW

**Comment by ist-qa** [comment]
Container built: istresearch/p4-engage:ci-4.0.15.1-P4-3944

**Comment by ist-qa** [comment]
Container built: istresearch/rapidpro:ci-4.0.15.1-P4-3944-generic

**Comment by ist-qa** [comment]
Container built: istresearch/p4-engage:pyapp-ci-4.0.15.1-P4-3944

**Comment by ist-qa** [comment]
Container built: istresearch/p4-engage:ci-4.0.15.1-P4-3944

**Comment by ist-qa** [comment]
Container built: istresearch/rapidpro:ci-4.0.15.1-P4-3944-generic

File: engage/channels/types/postmaster/apks.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+        #enddef post
+
+    #endclass PostmasterInfo
+
+    class DownloadPostmaster(LogExtrasMixin, OrgPermsMixin, BaseAPIView):
+        """
+        Endpoint for public access via random short-lived UUID in URL path to download pm APK.
+        """
+        permission_classes = (SSLorLocalTrafficPermission,)
+        permission = "apks.apk_list"
```
[the remark this answers is not in the document]
Comment (baracudda) [reply engage/channels/types/postmaster/apks.py:108]: public with a single caveat... `postoffice-only` as the UUID param needs to be non-public. so we need Django to actually use the middleware to give us a user account, if possible, so we can enforce restrictions after the fact.  hence why we cannot have `permissions = False` or removed entirely.  A really odd quirk, but that's how the framework operates.

[the remark this answers is not in the document]
Comment (rsdunya) [reply engage/channels/types/postmaster/apks.py:108]: Okay makes sense, I remember you speaking about that earlier but I didn't catch the full context of the conversation. Thanks for the clarification.

**Review by rsdunya — APPROVED** [review]

**Pull request merged**